teams/vsa/engineering/forms-system/readme.md

```diff
@@ -1,17 +1,50 @@
-## VSA internal working guidance on the use of VA Forms System (VAFS) and the VA fork of react-json-schema-form (RJSF)
-* [2020-03-forms-doc-feedback.md](./2020-03-forms-doc-feedback.md) \
-VSA front-end developers had an open discussion about issues they ran into while working with Forms. Here's the candid feedback. 
-  
-* [possible-issues.md](./possible-issues.md) \
-Our current (evolving) list of *possible* issues, things we may need to vett internally.
+# VAFS/RJSF Recipes & Best Practices
+
+The VA Forms System (VAFS) is built upon the VA fork of react-json-schema-form (RJSF). Here is VSA's set of internal guidelines and recipies for common yet more advanced forms you may need to create using VASF/RJSF. The recipies are organized by the form 'task' you are trying to achieve. 
+
+## Style and Organization
+
+- [Recommended Code Structure](./code_structure.md) - A way to organize your code so that it is easy to maintain.
+
+- [Using a Constants File](./constants.md) - Using a central file for all the variables you will use over and over again.
+
+
+## Implementation Patterns
+These real-world implementation patterns are somewhat gnarlky in detail. In some situations we may be using the forms system beyond what it was originally designed for.
+
+- [List Loop](./list_loop.md) - This is needed to build a form that allows a user to input a few items and a separate page for each of those items to enter more information about that respective item.
+
+- [Custom React Component: Simple Case](./custom_component_simple.md) - Update form state with an onChange handler within a custom React component.
 
-### Other (external) useful forms-related guidance:
+- [Custom React Component: Multiple/Complex](./custom_component_multiple.md) - Using a custom React component requires that you pass in the form's schema for that component (map a small slice of form state to the component). In this scenario, we use the same component twice and we pass in a different instance of the schema to each instance of the component. Since each instance only has access to its own individual schema/form state, we are using Redux in order for our instances to see each other's form state.

```

Comment:
@bradl3yC and @mr0sari0 how accurate is this statement? If we compare this scenario against the simple case above, does it explain why we have to peek into redux form state?

Reply:
Yeah this pretty much sums it up - we used a permanentAddress and a temporaryAddress - each had their own data and we needed to change a 3rd slice of state called currentAddress - this lived outside of them but they had it mapped to determine if their radio button was selected or not

Reply:
This is pretty accurate @short000 ! The simple case just outlines how you can use `updateFormData` to modify your formData. This scenario describes a need not mentioned in that case; the need being a way to share state between different schema properties.